python/taichi/examples/autodiff/diff_sph/diff_sph_demo.py

```diff
@@ -0,0 +1,632 @@
+import taichi as ti

```

Comment:
nit: can we remove the `demo` in the filename? ;)

Reply:
Sure. Have renamed and add some comments explaining what this demo is doing.